src/nwm_routing/src/nwm_routing/__main__.py

```diff
@@ -941,6 +953,23 @@ def get_waterbody_water_elevation(waterbodies_df, q0):
     return waterbodies_df
 
 
+def new_lastobs(run_results, time_increment):
+    df = pd.concat(
+        [
+            pd.DataFrame(
+                # TODO: Add time_increment (or subtract?) from time_since_lastobs
+                np.array([rr[3][1],rr[3][2]]).T,
+                index=rr[3][0],
+                columns=["time_since_lastobs", "lastobs_discharge"]
+            )
+            for rr in run_results
+            if not rr[3][0].size == 0
+        ],
+        copy=False,
+    )
+    df["time_since_lastobs"] = df["time_since_lastobs"] - time_increment
+    return df
+
 def main_v03(argv):
```

Comment:
This function could use some commenting to explain the sequence of processes... I'll suggest (based on my understanding) in comments below...

Reply:
Good suggestion. Expanded the doc-string with these details.